lib/pharos/addons/ingress-nginx/addon.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+Pharos.addon 'ingress-nginx' do
+  version '0.12.0'
+  license 'Apache License 2.0'
+
+  config {
+    attribute :configmap, Pharos::Types::Hash
+    attribute :node_selector, Pharos::Types::Hash
+    attribute :default_backend, Pharos::Types::Hash
+  }
+
+  config_schema {
+    optional(:configmap).filled(:hash?)
+    optional(:node_selector).filled(:hash?)
+    optional(:default_backend).schema {
+      optional(:image).filled(:str?)
+    }
+  }
+
+  install {
+    install(
```

Comment:
Overriding the name `install` for two different things here isn't good, this should probably be `install_stack`.

Reply:
totally agree, having two `install`s makes it really hard to grok out

Reply:
Ok, kinda agree that it's confusing.. but so is `apply_stack`. What is a stack (this is an addon) and why should user know about it?

Reply:
If we assume that addon is 1:1 with a stack, then I think the high-level API addon API shold look like:

```ruby
let(:configmap) { config.configmap || {} }
let(:node_selector) { config.node_selector || {} }
let(:image_name) { config.default_backend.image || IMAGES[cpu_arch.name] }`
...
```

Reply:
My suggestion is to just keep the `def install ...` for this PR, and then have a separate PR for making the DSL less `def`-ish.

Reply:
Is there something else "wrong" with the current `install` block except that we call also method with the same name? If no, then I would keep it here and just try to come up with better name.

Options:
- `apply_stack` (bad, as I said we don't have concept of stacks explained)
- `install` (bad, it has the same name as the block)
- `apply_resources` (better? we have thing called resources folder)
- `install_resources` (better? we have thing called resources folder)

Reply:
my vote would go to `apply_resources`, that maps it to kube API usage well too. we essentially `apply` bunch of `resources`

Reply:
```ruby
  install {
    install_resources(
      configmap: config.configmap || {},
      node_selector: config.node_selector || {},
      image: image_name,
      default_backend_replicas: default_backend_replicas
    )
  }
```

Reply:
It's a bit unclear what happens and when. 

If you have an addon without `install` I suppose it just does the resource applying.

We could have something like:


```ruby
hooks[:install] = proc do
  hooks.values_at(:before, :apply, :after).compact.map(&:call)
end
```

So normally you would then do:
```
before do
 puts "installing foofoo"
end

after do
  puts "you can use foofoo by foofooing"
end
```

If you don't have any resources or somehow want to mess with the applying you use:

```ruby
install do
  ssh { puts file('/etc/motd`).read }
end
```

Reply:
> If you have an addon without install I suppose it just does the resource applying.

Yes. We need to document this.

I think we can add before/after_install and before/after_uninstall later.